From 629744300e5ef6c4483f2787863da04037ca87bf Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 16 Jul 2014 20:42:37 -0400
Subject: [md] dm-mpath: remove map_io()

Message-id: <1405543364-20901-34-git-send-email-snitzer@redhat.com>
Patchwork-id: 85916
O-Subject: [RHEL7.1 PATCH 33/40] dm mpath: remove map_io()
Bugzilla: 1117872
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>

BZ: 1117872

Upstream commit 36fcffcc6500228efdfaf3a36761dd57a38366e3
Author: Hannes Reinecke <hare@suse.de>
Date:   Fri Feb 28 15:33:47 2014 +0100

    dm mpath: remove map_io()

    multipath_map() is now just a wrapper around map_io(), so we
    can rename map_io() to multipath_map().

    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>
    Reviewed-by: Jun'ichi Nomura <j-nomura@ce.jp.nec.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-mpath.c b/drivers/md/dm-mpath.c
index 1c3a941..c626feb 100644
--- a/drivers/md/dm-mpath.c
+++ b/drivers/md/dm-mpath.c
@@ -375,9 +375,13 @@ static int __must_push_back(struct multipath *m)
 
 #define pg_ready(m) (!(m)->queue_io && !(m)->pg_init_required)
 
-static int map_io(struct multipath *m, struct request *clone,
-    union map_info *map_context)
+/*
+ * Map cloned requests
+ */
+static int multipath_map(struct dm_target *ti, struct request *clone,
+    union map_info *map_context)
 {
+ struct multipath *m = (struct multipath *) ti->private;
  int r = DM_MAPIO_REQUEUE;
  size_t nr_bytes = blk_rq_bytes(clone);
  unsigned long flags;
@@ -909,17 +913,6 @@ static void multipath_dtr(struct dm_target *ti)
 }
 
 /*
- * Map cloned requests
- */
-static int multipath_map(struct dm_target *ti, struct request *clone,
-    union map_info *map_context)
-{
- struct multipath *m = (struct multipath *) ti->private;
-
- return map_io(m, clone, map_context);
-}
-
-/*
  * Take a path out of use.
  */
 static int fail_path(struct pgpath *pgpath)
-- 
1.7.1